Source/loadsave.cpp

```diff
@@ -1810,25 +1851,29 @@ void LoadGame(bool firstflag)
 
 	ViewPosition = { viewX, viewY };
 	ActiveMonsterCount = tmpNummonsters;
-	ActiveMissileCount = tmpNummissiles;
 	ActiveObjectCount = tmpNobjects;
 
 	for (int &monstkill : MonsterKillCounts)
 		monstkill = file.NextBE<int32_t>();
 
 	if (leveltype != DTYPE_TOWN) {
+		std::vector<int8_t> tmpActiveMissiles(MaxMissilesForSaveGame);

```

Comment:
This can be a `std::array`, it's never resized. Strictly speaking I don't think it's needed at all though. Missile IDs are very rarely used, if you clean up the few points where ID comparison/lookup is performed you can skip this completely and just insert from the start of the `Missiles` container.

Reply:
I used a `std::vector` so it's allocated on heap.
Cause some platforms have small stack size (like 3ds, see #3540).
But perhaps this is premature? 🤔

We could remove `tmpActiveMissiles`, if we don't want to have a stable id anymore.

Reply:
It looks like this is just 125 bytes. As long as it's under 32 KB and does no recursion, it should be fine.

Reply:
I changed it to normal stack arrays. 🙂